src/transferslot.cpp

```diff
@@ -227,7 +227,7 @@ TransferSlot::~TransferSlot()
 
         if (cachetransfer)
         {
-            transfer->client->transfercacheadd(transfer);
+            transfer->client->transfercacheadd(transfer, nullptr);
```

Comment:
Yes, I would much rather have made that a reference, but since database operations are being called from a destructor, we can't have a committer as a parameter (or any parameters at all, of course).   Requiring one as a parameter very useful in general I think, in order to remind ourselves that there are database operations going on in the function call, which otherwise is often not obvious.  And it helps a lot with finding the code paths that need committer objects too.   Though, passing them as parameters is not strictly needed, since the database operations will be processed whether they are in a transaction or not - it's just a matter of knowing and reminding ourselves to batch them in transactions for efficiency where possible.   So anyway, making the parameter a pointer instead of a reference here means that we can pass the object when it's available, but pass null from destructors.   So we are reminded that we should be making transactions, but we can also choose to side-step actually passing the transaction object when it's inconvenient to do so.   I hope that makes sense!  cheers